components/widgets/inputs/users_emails_input.tsx

```diff
@@ -243,34 +319,32 @@ export default class UsersEmailsInput extends React.PureComponent {
             }
         }
         if (action.action !== 'input-blur' && action.action !== 'menu-close') {
-            this.props.onInputChange(inputValue);
+            this.props.onInputChange(inputValue.value);
         }
     }
 
-    optionsLoader = (input, callback) => {
-        const customCallback = (options) => {
+    optionsLoader = (_: string, callback: (options: OptionsType<AppSelectOption>) => void ) => {
+        const customCallback = (options: OptionsType<AppSelectOption>) => {
             this.setState({options});
             callback(options);
         };
-        const result = this.props.usersLoader(this.props.inputValue, customCallback);
-        if (result && result.then) {
-            result.then(customCallback);
-        }
+        return this.props.usersLoader(this.props.inputValue, customCallback);
     }
 
-    showAddEmail = (input, values, options) => {
+    showAddEmail = (input: string, _:any, options: any[]): boolean => {
         return this.props.emailInvitationsEnabled && options.length === 0 && isEmail(input);
     }
 
     onFocus = () => {
-        this.selectRef.current.handleInputChange(this.props.inputValue, {action: 'custom'});
+        this.selectRef.current?.handleInputChange({value: this.props.inputValue}, {action: 'custom'});
```

Comment:
How come we're changing the object being passed in here?

Reply:
Up stream types were  not agreeing with the way these were structured. This happened several times, theses functions are basically saying that could take any of 2 or 3 types and it was making the type signatures really loopy

Reply:
I'm not sure I understand, so you're saying that the types were different than the ones we were passing in? Can you post the error you would get if you left it alone?